pennylane/ops/qubit/special_unitary.py

```diff
@@ -298,32 +434,240 @@ def compute_matrix(theta, num_wires):
+
+        # Compute the Omegas from the Jacobian. The adjoint of U(theta) is realized via -theta
+        U_dagger = self.compute_matrix(-_detach(theta, interface), num_wires)
+        # After contracting, move the parameter derivative axis to the first position
+        return qml.math.transpose(qml.math.tensordot(U_dagger, jac, axes=[[1], [0]]), [2, 0, 1])
+
+    def get_one_parameter_coeffs(self, interface):
+        r"""Compute the Pauli basis coefficients of the generators of one-parameter groups
+        that reproduce the partial derivatives of a special unitary gate.
 
         Args:
-            theta (tensor_like): Pauli coordinates of the exponent :math:`A(\theta)`
-            wires (Iterable[Any] or Wires): the wire(s) the operation acts on
-            num_wires (int): The number of wires
+            interface (str): The auto-differentiation framework to be used for the
+                computation.
+
+        Returns:
+            tensor_like: The generators for one-parameter groups that reproduce the
+            partial derivatives of the special unitary gate defined by ``theta``.
+            There are :math:`d=4^n-1` generators for :math:`n` qubits, so that the
+            output shape is ``(4**num_wires-1, 2**num_wires, 2**num_wires)``.
+
+        Given a generator :math:`\Omega` of a one-parameter group that reproduces a partial derivative
+        of a special unitary gate, it can be decomposed in the Pauli basis of :math:`\mathfrak{su}(N)`
+        via
+
+        .. math::
+
+            \Omega = \sum_{m=1}^d \omega_m P_m
+
+        where :math:`d=4^n-1` is the size of the basis for :math:`n` qubits and :math:`P_m` are the
+        Pauli words making up the basis. As the Pauli words are orthonormal with respect to the
+        `trace or Frobenius inner product <https://en.wikipedia.org/wiki/Frobenius_inner_product>`__
+        (rescaled by :math:`2^n`), we can compute the coefficients using this
+        inner product (:math:`P_m` is Hermitian, so we skip the adjoint :math:`{}^\dagger`):
+
+        .. math::
+
+            \omega_m = \frac{1}{2^n}\operatorname{tr}\left[P_m \Omega \right]
+
+        The coefficients satisfy :math:`\omega_m^\ast=-\omega_m` because :math:`\Omega` is
+        skew-Hermitian. Therefore they are purely imaginary.
+
+        .. warning::
+
+            An auto-differentiation framework is required by this function.
+            The matrix exponential is not differentiable in Autograd. Therefore this function
+            only supports JAX, Torch and Tensorflow.
+
+        ..seealso:: `~.transforms.insert_paulirot.get_one_parameter_generators`
+
+        """
+        num_wires = self.hyperparameters["num_wires"]
+        basis = pauli_basis_matrices(num_wires)
+        generators = self.get_one_parameter_generators(interface)
+        return qml.math.tensordot(basis, generators, axes=[[1, 2], [2, 1]]) / 2**num_wires
+
+    def decomposition(self):
```

Comment:
Usually `decompositions` simply call `compute_decomposition` with the parameters. Why we do not follow this logic here and update `compute_decompositions` instead?

Reply:
This was suggested by @AlbertMitjans, questioning the use of static methods in the operator interface more generally. Maybe we can have a discussion on this? 
Here it convinced me because it allows access of `get_one_parameter_coeffs` and the operation's matrix via `self`, rather than using `qml.SpecialUnitary.get_one_parameter_coeffs` and `qml.SpecialUnitary.compute_matrix`.

Reply:
Related: https://github.com/PennyLaneAI/pennylane/pull/3650/#discussion_r1091741622

Reply:
Thanks, great that you went with `decompositions` :+1: